plugins/inputs/postgresql/postgresql.go

```diff
@@ -155,7 +155,11 @@ func (p *Postgresql) accRow(row scanner, acc telegraf.Accumulator, columns []str
 	}
 	if columnMap["datname"] != nil {
 		// extract the database name from the column map
-		dbname.WriteString((*columnMap["datname"]).(string))
+		if dbNameStr, ok := (*columnMap["datname"]).(string); ok {
+			dbname.WriteString(dbNameStr)
+		} else {
+			dbname.WriteString("postgres_global")
```

Comment:
Let's add a comment to point to some pg docs about how this was added in pg 12 and maybe some telegraf docs to tell the user when to expect it.